CLAUDE.md

```diff
@@ -0,0 +1,271 @@
+
+### Build Modes
+- **Standalone**: Pure NewDot application (web/desktop)
+- **HybridApp**: Combined OldDot + NewDot (mobile apps)
+- Controlled via `STANDALONE_NEW_DOT` environment variable
+
+## Core Architecture & Structure
+
+### Entry Points
+- `src/App.tsx`: Main application component with provider hierarchy
+- `src/Expensify.tsx`: Core application logic and initialization
+- `src/HybridAppHandler.tsx`: Manages HybridApp transitions and authentication
+- `index.js`: React Native entry point
+
+### Provider Architecture
+The application uses a nested provider structure for context management:
+1. **SplashScreenStateContextProvider**: Manages splash screen visibility
+2. **InitialURLContextProvider**: Handles deep linking
+3. **ThemeProvider**: Theme management
+4. **LocaleContextProvider**: Internationalization
+5. **OnyxListItemProvider**: Data layer provider
+6. **SafeAreaProvider**: Device safe areas
+7. **PopoverContextProvider**: Global popover state
+8. **KeyboardProvider**: Keyboard state management
+
+### Data Layer
+- **Onyx**: Custom data persistence layer for offline-first functionality
+- **ONYXKEYS.ts**: Centralized key definitions for data store
+- Supports optimistic updates and conflict resolution
+
+## Key Features & Modules
+
+### Core Functionality
+1. **Expense Management**
+   - Receipt scanning and SmartScan
+   - Expense creation and editing
+   - Distance tracking
+   - Per diem support
+   - Split expenses
+
+2. **Reporting**
+   - Report creation and submission
+   - Approval workflows
+   - Report fields and custom attributes
+   - Bulk operations
+
+3. **Workspace/Policy Management**
+   - Policy creation and configuration
+   - Member management
+   - Categories, tags, and tax rates
+   - Accounting integration settings
+   - Approval workflows
+
+4. **Travel**
+   - Trip management
+   - Travel booking integration
+   - Travel policy enforcement
+
+5. **Search & Filtering**
+   - Advanced search with filters
+   - Saved searches
+   - Search parser (Peggy-based)
+
+6. **Payment & Cards**
+   - Expensify Card management
+   - Bank account connections (Plaid)
+   - Payment methods
+   - Company cards integration
+   - Wallet functionality
+
+7. **Accounting Integrations**
+   - QuickBooks Online
+   - Xero
+   - NetSuite
+   - Sage Intacct
+   - QuickBooks Desktop
+   - Generic accounting connections
+
+8. **Communication**
+   - Chat functionality
+   - Task management
+   - Mentions and notifications
+   - Thread organization
+
+9. **Invoice Management**
+   - Invoice creation and sending
+   - Invoice rooms
+
+## Navigation & Routing
+
+### Structure
+- `src/SCREENS.ts`: Screen name constants
+- `src/ROUTES.ts`: Route definitions and builders
+- `src/NAVIGATORS.ts`: Navigator configuration
+
+### Key Navigators
+- **ProtectedScreens**: Authenticated app screens
+- **PublicScreens**: Login and onboarding
+- **RHP (Right Hand Panel)**: Settings and details panels
```

Comment:
(NAB) I always thought it was Right Hand Pane? 😅 Maybe either is valid

Reply:
Yeah that seems correct, thanks!

Reply:
Lol actually i see people using both. I'll just mention both in the rules